src/Chapter2.hs

```diff
@@ -605,8 +621,9 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
-
+duplicate [] = []
+duplicate [e] = [e, e]
+duplicate (h : rest) = h : h : duplicate rest
```

Comment:
Awesome ⭐ 

You can notice that the `[e]` case is actually covered by `(h : rest)` with the rest = `[]`, as the list can always be represented like this`[e] = e : []` 🙂

Reply:
Got it. Nice suggestions. Thanks